### ISSUE TITLE
Does not work with server-side Blazor in 3.0.0-preview7

### ISSUE THREAD

**Issue by noahc3** [cloudcrate/BlazorStorage#14]
When attempting to use this package with a server-side rendered Blazor application, the page fails to render when the package tries to cast the `RemoteJSRuntime` instead of the client one (I assume).

Stack Trace:

```
System.InvalidCastException: Unable to cast object of type 'Microsoft.AspNetCore.Components.Server.Circuits.RemoteJSRuntime' to type 'Microsoft.JSInterop.IJSInProcessRuntime'.
   at Cloudcrate.AspNetCore.Blazor.Browser.Storage.StorageBase..ctor(IJSRuntime jsRuntime)
   at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.AspNetCore.Components.ComponentFactory.<>c__DisplayClass6_0.<CreateInitializer>b__2(IComponent instance)
   at Microsoft.AspNetCore.Components.ComponentFactory.PerformPropertyInjection(IComponent instance)
   at Microsoft.AspNetCore.Components.ComponentFactory.InstantiateComponent(Type componentType)
   at Microsoft.AspNetCore.Components.Rendering.Renderer.InstantiateChildComponentOnFrame(RenderTreeFrame& frame, Int32 parentComponentId)
```

Edit: Reduced stacktrace

**Comment by LunicLynx** [comment]
Intresting, i guess it is not really considered to be used in the server-side setup. But could be possible with RemoteJsRuntime, if i understand correctly.

What is your use case if i can ask?

Normally you would have far better options in a server-side setup than local storage.

**Comment by noahc3** [comment]
Just wanted to track a theme setting in localstorage so far. Doing it through JavaScript is trivial but ideally I'd like as little manual JSInterop as possible. 

I'm mostly making stuff with blazor without any form of client identification ex. user accounts, so storing settings on the client is necessary.

**Comment by konradbartecki** [comment]
I've fixed it for server-side blazor in this PR
https://github.com/cloudcrate/BlazorStorage/pull/15
You can download the project and simply reference it until nuget is ready, there is an example available that's doing exactly that.

**Comment by LunicLynx** [comment]
This is fixed with Version `3.0.0-preview8.19405.7.1`on nuget. Thanks to @konradbartecki!
